src/arrays.jl

```diff
@@ -217,6 +217,11 @@ function ndims(aop::ArrayOp)
     length(aop.output_idx)
 end
 
+function SymbolicUtils.substitute(x::ArrayOp, dict; fold=true)
+  out_idx, expr, redop, ter, tail... = arguments(x)
+  s = x -> substitute(x, dict, fold=fold)
+  return operation(x)((map(s, out_idx)...,), s(expr), redop, s(ter), tail...)
+end
```

Comment:
You may want to check that the `operation(x)` is ArrayOp, I changed the interface in #495 ...

Reply:
The original `substitute` should work out in the other case...